beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -123,14 +126,67 @@ func (bs *BeaconChainServer) ListValidatorBalances(
 	return &ethpb.ValidatorBalances{Balances: res}, nil
 }
 
-// GetValidators retrieves the current list of active validators.
+// GetValidators retrieves the current list of active validators with an optional historical epoch flag to
+// to retrieve validator set in time.
 //
-// The request may include an optional historical epoch to retrieve a
-// specific validator set in time.
+// TODO(#3045): Implement validator set for a specific epoch. Current implementation returns latest set,
+// this is blocked by DB refactor.
 func (bs *BeaconChainServer) GetValidators(
-	ctx context.Context, req *ethpb.GetValidatorsRequest,
-) (*ethpb.Validators, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	ctx context.Context,
+	req *ethpb.GetValidatorsRequest) (*ethpb.Validators, error) {
+
+	validators, err := bs.beaconDB.Validators(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not retrieve validators: %v", err)
+	}
+	totalSize := len(validators)
+
+	// If options PageToken and PageSize are not specified, return the validator list capped at DefaultPageSize.
+	if req.PageToken == "" && req.PageSize == 0 {
```

Comment:
Instead of having a different code path, how about just setting these?

```go
if req.PageToken == "" {
  req.PageToken = "0"
}
if req.PageSize = 0 {
  req.PageSize = params.BeaconConfig().DefaultPageSize
}
// Then continue with the same logic.
```

This will avoid bugs like the one on line 153 where next page should have been 1 and you have 0.

Reply:
This is better!